esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3271,8 +3271,19 @@ void WsWuHelpers::copyWsWorkunit(IEspContext &context, IWorkUnit &wu, const char
 
     queryExtendedWU(&wu)->copyWorkUnit(src, false, false);
 
-    SCMStringBuffer token;
-    wu.setSecurityToken(createToken(wu.queryWuid(), context.queryUserId(), context.queryPassword(), token).str());
+    //Digitally sign the workunit
+    if (!wu.setSecuritySignature())
+    {
+        if (isEmptyString(context.queryPassword()))
```

Comment:
This brings up a good question.  Are we changing the ownership to the current user whenever we sign the workunit?  We should evaluate the scenarios.  Unless that is happening above these lines now then I THINK you are trying to sign based on the owner of the workunit but then here you want to check the username of the current user.  Those aren't the same thing (unless it's always updated).  In any case you are actually checking the password here which isn't used to generate the signature.

Reply:
The purpose of line 3277 is that if I cannot sign the workunit, I need to make sure there is a password in order to create a legacy token

Reply:
@afishbeck I am not sure what you mean about changing ownership. I am creating the token using the same user as was the legacy token (line 3284).  Can you elaborate?

Reply:
@RussWhitehead mostly I was noticing that you try and resign the workunit based on the original OWNER, but hten if you create a security token below, it is based on the current USER.  So permission checking is very inconsistent in this case between signed and tokened workunits.

I think the old behavior where the token is based on the current user is more correct.  I.e. if I copy a workunit to my environment it should end up with my permissions, not that of the original owner.  But that is open for discussion.